src/main/python/janusgraph_python/driver/ClientBuilder.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2018 JanusGraph Python Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from ..structure.io.GraphsonReader import JanusGraphSONReader
+from ..structure.io.GraphsonWriter import JanusGraphSONWriter
+
+from gremlin_python.driver.driver_remote_connection import DriverRemoteConnection
+from gremlin_python.structure.io.graphsonV3d0 import GraphSONReader
+from gremlin_python.structure.io.graphsonV3d0 import GraphSONWriter
+
+
+class JanusGraphClient(object):
+    """
+        JanusGraph Client Builder which adds the Serializers for JanusGraph specific objects, predicates etc.
+    """
+
+    REMOTE_CONNECTION = None
+
+    def __init__(self, version=3.0):
+        """
+            Initializing with GraphSON version 3.0
+        Args:
+            version (int):
+        """
+
+        self.graphsonVersion = version
+        pass
+
+    def connect(self, url="loclahost", port="8182", graph="g", **kwargs):
+        """
+            Connect to JanusGraph's gremlin-server instance. Takes URL, Port and Graph
+
+        Args:
+            url (str): The URL of JanusGraph gremlin-server instance. Default to localhost
+            port (str): The PORT of JanusGraph gremlin-server instance. Default to 8182.
+            graph (str): The GraphTraversalSource being exposed from gremlin-server instance. Defaults to g
+
+        Keyword Args:
+            graphson_reader (GraphSONReader): GraphSONReader object with required Deserializers registered
+            graphson_writer (GraphSONWriter): GraphSONWriter object with required Serializers registered
+
+        Raises:
+            AttributeError: When invalid Keyword arguments is provided. T
+            he expected key needs to be `graphson_reader` and `graphson_writer`.
+
+        Returns:
+            JanusGraphClient
+        """
+
+        URL = "ws://{}:{}/gremlin".format(url, port)
+
+        if kwargs is None:
+
+            graphson_reader = JanusGraphSONReader().build()
+            graphson_writer = JanusGraphSONWriter().build()
+
+        else:
+            if "graphson_reader" in kwargs and "graphson_writer" in kwargs:
+                graphson_reader = kwargs["graphson_reader"]
+                graphson_writer = kwargs["graphson_writer"]
+            else:
+                raise AttributeError("Additional parameters if provided needs to be keywords arguments of "
```

Comment:
I setup a connection with url, port, and graph and get an error that I need to set graphson_reader and graphson_writer even though it looks like kwargs should be None.
 
```
connection = JanusGraphClient().connect(url="0.0.0.0", port="8182", graph="airroutes_traversal").get_connection()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/janusgraph_python/driver/ClientBuilder.py", line 74, in connect
    raise AttributeError("Additional parameters if provided needs to be keywords arguments of "
AttributeError: Additional parameters if provided needs to be keywords arguments of `graphson_reader` and `graphson_writer`
```

Reply:
@chupman : Are you on latest code? I remember having that bug, and I had pushed fix resolving the same on this PR. The latest code on fix#3 branch has fixed this issue. I was able to get the following thing working, without any error:

```
connection = JanusGraphClient().connect(url="13.71.86.xxx", port="8182", graph="g").get_connection()
g = Graph().traversal().withRemote(connection)
print("Count: ", g.V().count().next())
```

Reply:
I built it last week so it should be the most current, but I'll try to build it again to validate.

Reply:
I'm going to try with a fresh Ubuntu VM, but so far on my CentOS VM and my macbook I'm unable to build the library today.

if I run ./build=.sh, the command will complete, but the library is never built
```
Building library
./build.sh: line 65: ./build-library.sh: Permission denied
```

If I use pyb 
```
[root@some-vm janusgraph-python]# pyb
PyBuilder version 0.11.17
Build started at 2018-10-11 13:15:17
------------------------------------------------------------
[INFO]  Building janusgraph_python version 0.0.9
[INFO]  Executing build in /root/janusgraph-python
[INFO]  Going to execute tasks: clean, install_dependencies, prepare, compile_sources, package, publish
[INFO]  Removing target directory /root/janusgraph-python/target
[INFO]  Installing all dependencies
[INFO]  Running unit tests
[WARN]  unittest_file_suffix is deprecated, please use unittest_module_glob
[INFO]  Executing unit tests from Python modules in /root/janusgraph-python/src/unittest/python
[INFO]  Executed 10 unit tests
[ERROR] Test has error: unittest.loader._FailedTest.RelationIdentifierSerializer_test
[ERROR] Test has error: unittest.loader._FailedTest.RelationIdentifierDeserializer_test
[ERROR] Test has error: unittest.loader._FailedTest.CircleSerializer_test
[ERROR] Test has error: unittest.loader._FailedTest.GeoShapeDeserializer_test
[ERROR] Test has error: unittest.loader._FailedTest.PointSerializer_test
[ERROR] Test has error: unittest.loader._FailedTest.RelationIdentifier_test
[ERROR] Test has error: unittest.loader._FailedTest.Circle_test
[ERROR] Test has error: unittest.loader._FailedTest.Point_test
[ERROR] Test has error: unittest.loader._FailedTest.GraphsonReader_test
[ERROR] Test has error: unittest.loader._FailedTest.GraphsonWriter_test
------------------------------------------------------------
BUILD FAILED - There were 10 error(s) and 0 failure(s) in unit tests
------------------------------------------------------------
Build finished at 2018-10-11 13:15:19
Build took 2 seconds (2040 ms)
[root@some-vm janusgraph-python]# git status
# On branch fix_#3
```

Reply:
@chupman : I just spawned off a fresh Ubuntu VM, and I was able to make it work. I have tested it just on Windows machines till now. I'll test with Ubuntu and let me know. 

As for testing on CentOS, well I plan on downloading a base CentOS docker image, pulling repo inside and testing it. This is a real strange issue, and don't know why all tests are failing on your end. Can you do `pyb -v` in verbose mode so that I can check the logs?

This is a real nice thing you are doing. We ideally would want to test in on all OSs possible before releasing. In meantime if you can get logs for `pyb -v` that will be great

Reply:
@chupman : Looks like you are getting permission denied on `build.sh`. Did you first do `chmod +x build.sh` to give it exec permissions?

Somehow I suspect that the shell script in your case is picking up default Python, i.e. python2 not Pythn3. Note that the libs aren't supported on python2.

I build the library fresh from scratch on my most updated repo (debasishdebs/janusgraph-py) and it was failing initially, coz it was picking up the default python2 not 3. Reason being python 2 provides `__cmp()__` to assert equality while that was changes to `__eq()__ and __neq()__` in python3. 

I'll have better idea wen you can do following things:
- Print the whole log of ./build.sh -d true -b true -i true. I do print the python version being used each time before building docs and before building lib. That will help accertain if the error is because of Python versions.
- If Python version is still 3.X, then run just `pyb -v` instead of `pyb` which is in verbose mode to have better idea why build is failing.

If its the 1st one, as per my suspicion, then I've a commit ready. Its already pushed to my personal repo (janugraph-py). If the reason is confirmed, I can push the commit to this PR and update the build scripts accordingly.

The Python version is printed somewhat in similar way as follows:
![image](https://user-images.githubusercontent.com/1013906/47081538-8794c980-d228-11e8-92f2-478e8811309b.png)

Reply:
On my Mac I ran into a different issue. I'm not sure if pybuilder is supposed to link or copy Geo.py after it creates the GeoPredicate folder

```  
File "/Users/chupman/github/janusgraph-python/src/main/python/janusgraph_python/core/attribute/__init__.py", line 21, in <module>
    from .GeoPredicate import Geo
ImportError: cannot import name 'Geo'
```

On CentOS
It is defaulting to python 2. I was able to get it to run with python3, but ran into different errors. I put the output in a [GIST](https://gist.github.com/chupman/e5007fec67fa4b14cc3880d603c25a14)

Reply:
@sjudeng @chupman : The build failures were part of me incorporating few of comments in this PR. My bad. I got access to few Ubuntu and CentOS machines, and I've updated this PR with the fixes for the build fauilres. I just tested on Ubuntu and CentOS and both works file, can you guys confirm the same?

Also, the usage of build script has changed to following: 
```/build.sh -d true -b true -i true -p 3.6```
Where the parameter -p corresponds to python version you have installed on your system. It can be verified by running either of following commands python3/python3.4/python3.5/python3.6 whichever is installed, you pass accordingly. I've updated the docs for the same.

Reply:
pulled the changes and it built fine on my mac. Here's the output with the test warnings.

Build finished at 2018-10-19 19:26:51
Build took 4 seconds (4497 ms)
Building library
Python 3.6.6
PyBuilder version 0.11.17
Build started at 2018-10-19 19:26:51
------------------------------------------------------------
[INFO]  Building janusgraph_python version 0.0.9
[INFO]  Executing build in /Users/chupman/github/janusgraph-python
[INFO]  Going to execute tasks: clean, install_dependencies, prepare, compile_sources, package, publish
[INFO]  Removing target directory /Users/chupman/github/janusgraph-python/target
[INFO]  Installing all dependencies
[INFO]  Processing batch dependency 'gremlinpython==3.3.3'
[INFO]  Running unit tests
[WARN]  unittest_file_suffix is deprecated, please use unittest_module_glob
[INFO]  Executing unit tests from Python modules in /Users/chupman/github/janusgraph-python/src/unittest/python
[INFO]  Executed 19 unit tests
[INFO]  All unit tests passed.
[INFO]  Building distribution in /Users/chupman/github/janusgraph-python/target/dist/janusgraph_python
[INFO]  Copying scripts to /Users/chupman/github/janusgraph-python/target/dist/janusgraph_python/scripts
[INFO]  Writing setup.py as /Users/chupman/github/janusgraph-python/target/dist/janusgraph_python/setup.py
[INFO]  Collecting coverage information
[INFO]  Running unit tests
[WARN]  unittest_file_suffix is deprecated, please use unittest_module_glob
[INFO]  Executing unit tests from Python modules in /Users/chupman/github/janusgraph-python/src/unittest/python
[INFO]  Executed 19 unit tests
[INFO]  All unit tests passed.
[WARN]  Module 'janusgraph_python.core.attribute.GeoPredicate.Geo' was not imported by the covered tests
[WARN]  Coverage for module 'janusgraph_python.core.attribute.GeoPredicate.Geo' cannot be established - module failed: No module named 'janusgraph_python.core.attribute.datatypes'
[WARN]  Module 'janusgraph_python.core.attribute.TextPredicate.Text' was not imported by the covered tests
[WARN]  Module 'janusgraph_python.driver' was not imported by the covered tests
[WARN]  Branch coverage below 60% for janusgraph_python.core.datatypes.RelationIdentifier: 50%
[WARN]  Test coverage below 50% for janusgraph_python.driver.ClientBuilder:  0%
[WARN]  Branch coverage below 60% for janusgraph_python.driver.ClientBuilder:  0%
[INFO]  Overall coverage is 84%
[INFO]  Overall coverage branch coverage is 62%
[INFO]  Overall coverage partial branch coverage is 83%
[INFO]  Building binary distribution in /Users/chupman/github/janusgraph-python/target/dist/janusgraph_python
------------------------------------------------------------
BUILD SUCCESSFUL